warehouse/ingest-core/src/main/java/datawave/ingest/mapreduce/job/TableSplitsCache.java

```diff
@@ -45,8 +50,36 @@
     private static final short DEFAULT_MAX_SPLIT_DECREASE = 42;
     private static final double DEFAULT_MAX_SPLIT_PERCENTAGE_DECREASE = .5;
     private static final boolean DEFAULT_REFRESH_SPLITS = true;
+    
+    private AccumuloHelper cbHelper = null;
     private Path splitsPath = null;
     private Map<String,List<Text>> splits = null;
+    private Map<String,Map<Text,String>> splitLocations = null;
+    
+    private Map<Text,String> getSplitsWithLocation(String table) throws AccumuloException, AccumuloSecurityException, TableNotFoundException {
+        SortedMap<KeyExtent,String> tabletLocations = new TreeMap<>();
+        SortedMap<Text,String> tabletLocationsByEndRow = new TreeMap<>();
+        
+        MetadataServicer.forTableName(getClientContext(), table).getTabletLocations(tabletLocations);
+        for (Map.Entry<KeyExtent,String> entry : tabletLocations.entrySet()) {
+            // Note that if the tablet is currently in transition, then the "location" (entry.getValue()) will be null
+            String value = entry.getValue();
+            if (null == value) {
+                if (log.isDebugEnabled()) {
+                    log.debug("tablet location for " + entry.getKey() + " is null");
+                }
+                continue;
```

Comment:
curious whether the caller or the reader of the file wants to know about the split even if the location is empty at the time....